### PR TITLE
Add support for optional --space argument

### PR DESCRIPTION
**Pull request by AlexanderCurtin**
Adds an optional --space option which accepts the name of an octopus space.

# Changes
* bin/commands/options.js - added new --space parameter
* bin/commands/* - call new `ensureSpaceIsSet` command. (error handling is just allowing exception to be thrown)
* lib/api/space.js - Add `findAll` which just calls `/spaces/all`. There is no getter that takes a name in the octo api
* lib/commands/ensure-space-set - Accepts an optional space name, and if it's set, fetches and updates the api config. Errors otherwise.

### CODE REVIEW

File: bin/commands/octopack.js [hunk 1]
```diff
@@ -20,9 +21,10 @@ const builder = yargs =>
     .version(false)
 
 const handler = async args => {
-  const { host, apiKey, packageName, packageVersion, globs, base, replace, zip } = args
+  const { host, apiKey, space, packageName, packageVersion, globs, base, replace, zip } = args
 
   setApiConfig({ host, apiKey })
+  ensureSpaceIsSet.execute(space)
```
Comment (AlexanderCurtin) [inline bin/commands/octopack.js]: `await`

**Comment by AlexanderCurtin** [comment]
Sorry, kind of threw this together over the weekend, and noticing that the imported method/file names should match. More importantly I dropped some unawaited promises in there. I'm lucky that worked at all.

**Comment by parkerholladay** [comment]
I apologize for this sitting forever. I like the changes. I wish there was a way we could just integrate it in with the `setApiConfig()`. While you technically could, I like that you took the extra step to ensure that the space exists before adding it to the api host.

**Comment by parkerholladay** [comment]
For anyone still depending on this, I'll do one last release, but it's time to start promoting the official cli tool Octopus Deploy offers.

**Pull request merged**